src/Components/Shared/src/WebEventData.cs

```diff
@@ -60,28 +85,35 @@ private WebEventData(int browserRendererId, ulong eventHandlerId, EventFieldInfo
 
         public EventArgs EventArgs { get; }
 
-        private static EventArgs ParseEventArgsJson(Renderer renderer, JsonSerializerOptions jsonSerializerOptions, ulong eventHandlerId, string eventName, string eventArgsJson)
+        private static EventArgs ParseEventArgsJson(
+            Renderer renderer,
+            IWebEventJsonSerializerContext jsonSerializerContext,
+            ulong eventHandlerId,
+            string eventName,
+            string eventArgsJson)
         {
             try
             {
-                if (TryDeserializeStandardWebEventArgs(eventName, eventArgsJson, out var eventArgs))
+                if (TryDeserializeStandardWebEventArgs(eventName, eventArgsJson, jsonSerializerContext, out var eventArgs))
                 {
                     return eventArgs;
                 }
 
                 // For custom events, the args type is determined from the associated delegate
                 var eventArgsType = renderer.GetEventArgsType(eventHandlerId);
-                return (EventArgs)JsonSerializer.Deserialize(eventArgsJson, eventArgsType, jsonSerializerOptions)!;
+                return (EventArgs)JsonSerializer.Deserialize(eventArgsJson, eventArgsType, jsonSerializerContext.Options)!;
```

Comment:
Why's an options instance passed here instead of the context?

Reply:
In this case, the types being deserialized are user types and don't appear in the Blazor's internal context. Having to flow the user's context here would have been one of the sticking points in this API since a user would be responsible for ensuring arbitrary events that appear in their app as well as arbitrary libraries they reference are present in the context they can provide to Blazor.

Reply:
> a user would be responsible for ensuring arbitrary events that appear in their app as well as arbitrary libraries they reference are present in the context they can provide to Blazor

For users in the ecosystem that are motivated enough, could we encourage library authors expose context types for events that they define, and then users could union their context types either manually or with a new API that we can provide?

Reply:
Also, if the event types are public, then users could indicate them as `JsonSerializable` in their own assemblies and the gen'd context  would contain all the needed event types. This way there's no reliance of lib authors exposing their context type soon after .NET 6.0.

Reply:
We'll work on custom events as a follow up.